packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -33,7 +37,12 @@ const POPOVER_PROPS = {
 };
 
 function CopyMenuItem( { blocks, onCopy } ) {
-	const ref = useCopyToClipboard( () => serialize( blocks ), onCopy );
+	const ref = useCopyToClipboard( () => {
+		blocks = blocks.map( ( block ) =>
+			__experimentalCloneSanitizedBlock( block )
+		);
+		return serialize( blocks );
```

Comment:
Have you explored whether we could move `__experimentalCloneSanitizedBlock` to the `serialize` method? I see a similar logic together in another file in this PR. The way it works now is fine but we will have to ensure that we use the same logic any time we add new functionality or what’s worse when plugins use the current public API. I don’t know if that’s possible but it would be a nice improvement for developers.

Reply:
I see your point, but I don't think we can move the logic up any higher. If we move `__experimentalCloneSanitizeBlock` into `serialize`, we would end up stripping out internal attributes on much more than duplication, every time the post is reloaded.

In this file and in the copy-handler, I think it makes sense that when you want to make a copy of a block you should explicitly clone it before serializing. There is some danger that this could be missed with new functionality that involves block duplication, but I can't think of a way of enforcing it.